core/vm/eips.go

```diff
@@ -144,3 +145,22 @@ func enable2929(jt *JumpTable) {
 	jt[SELFDESTRUCT].constantGas = params.SelfdestructGasEIP150
 	jt[SELFDESTRUCT].dynamicGas = gasSelfdestructEIP2929
 }
+
+// enable3198 applies EIP-3198 (BASEFEE Opcode)
+// - Adds an opcode that returns the current block's base fee.
+func enable3198(jt *JumpTable) {
+	// New opcode
+	jt[BASEFEE] = &operation{
+		execute:     opBaseFee,
+		constantGas: GasQuickStep,
+		minStack:    minStack(0, 1),
+		maxStack:    maxStack(0, 1),
+	}
+}
+
+// opBaseFee implements BASEFEE opcode
+func opBaseFee(pc *uint64, interpreter *EVMInterpreter, scope *ScopeContext) ([]byte, error) {
+	baseFee, _ := uint256.FromBig(interpreter.evm.Context.BaseFee)
```

Comment:
This crashes if someone executes the BASEFEE opcode before the hf block. In `core/vm.go` Context.BaseFee is set to the uninitialized `baseFee` if no basefee was specified. This causes a sigsegv here during `uint256.FromBig`.
Repro:
```go
package main

import (
	"math/big"
	"github.com/holiman/uint256"
)

func main() {
	var baseFee *big.Int
	bf,_ := uint256.FromBig(baseFee) // produces Sigsegv
	_ = bf
}
```
This is of course not a very practical flaw, as it requires the opcode to be activated before the baseFee is set, but I think we should still not panic.
My suggestion would be to initialize baseFee in the context to `big.NewInt(0)` not nil, even before the hf block.

Reply:
Fixed!